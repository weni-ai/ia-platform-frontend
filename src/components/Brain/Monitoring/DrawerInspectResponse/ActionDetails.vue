<template>
  <section
    data-testid="action-details-container"
    class="action-details"
  >
    <UnnnicAvatarIcon
      data-testid="action-icon"
      class="action__icon"
      size="sm"
      icon="bolt"
      scheme="aux-blue"
    />
    <p
      data-testid="action-name"
      class="action__name"
    >
      {{ action.name }}
    </p>
  </section>
</template>

<script setup>
defineProps({
  action: {
    type: Object,
    required: true,
  },
});
</script>

<style scoped lang="scss">
.action-details {
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
  border-radius: $unnnic-border-radius-sm;

  padding: $unnnic-spacing-xs;

  display: flex;
  align-items: center;
  gap: $unnnic-spacing-xs;

  .action__name {
    color: $unnnic-color-neutral-dark;
    font-size: $unnnic-font-size-body-gt;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }
}
</style>
