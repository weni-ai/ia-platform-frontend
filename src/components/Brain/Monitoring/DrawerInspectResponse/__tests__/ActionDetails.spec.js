import { shallowMount } from '@vue/test-utils';
import { describe, it, expect, beforeEach } from 'vitest';

import ActionDetails from '../ActionDetails.vue';

describe('ActionDetails.vue', () => {
  let wrapper;
  const actionMock = {
    name: 'Sample Action',
  };

  beforeEach(() => {
    wrapper = shallowMount(ActionDetails, {
      props: {
        action: actionMock,
      },
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders the action name', () => {
    const actionName = wrapper.find('[data-testid="action-name"]');
    expect(actionName.exists()).toBe(true);
    expect(actionName.text()).toBe(actionMock.name);
  });

  it('should renders the action icon with correct props', () => {
    const avatarIcon = wrapper.findComponent('[data-testid="action-icon"]');
    expect(avatarIcon.exists()).toBe(true);

    expect(avatarIcon.props()).toMatchObject({
      size: 'sm',
      icon: 'bolt',
      scheme: 'aux-blue',
    });
  });

  it('should applies the correct class to the action details container', () => {
    const actionDetailsContainer = wrapper.find(
      '[data-testid="action-details-container"]',
    );
    expect(actionDetailsContainer.classes()).toContain('action-details');
  });
});
