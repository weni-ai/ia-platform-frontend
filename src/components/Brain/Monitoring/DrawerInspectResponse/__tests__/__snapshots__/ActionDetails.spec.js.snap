// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionDetails.vue > matches the snapshot 1`] = `
"<section data-v-2d5e21e2="" data-testid="action-details-container" class="action-details">
  <anonymous-stub data-v-2d5e21e2="" enabled="true" icon="bolt" size="sm" scheme="aux-blue" filled="false" opacity="true" data-testid="action-icon" class="action__icon"></anonymous-stub>
  <p data-v-2d5e21e2="" data-testid="action-name" class="action__name">Sample Action</p>
</section>"
`;
